nms/app/packages/magmalte/app/components/context/RefreshContext.js

```diff
@@ -0,0 +1,161 @@
+/**
+ * Copyright 2020 The Magma Authors.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * @flow strict-local
+ * @format
+ */
+'use strict';
+
+import {FetchEnodebs, FetchGateways} from '../../state/lte/EquipmentState';
+import {
+  FetchSubscriberState,
+  FetchSubscribers,
+} from '../../state/lte/SubscriberState';
+import {useContext, useEffect, useState} from 'react';
+
+export const REFRESH_INTERVAL = 30000;
+
+type Props = {
+  context: typeof React.Context,
+  networkId: string,
+  type: refreshType,
+  interval?: number,
+  id?: string,
+  enqueueSnackbar?: (msg: string, cfg: {}) => ?(string | number),
+  refresh: boolean,
+  lastRefreshTime?: string,
+};
+
+type refreshType = 'subscriber' | 'gateway' | 'enodeb';
+
+export function useRefreshingContext(props: Props) {
+  const ctx = useContext(props.context);
```

Comment:
Why did you change the original logic of refreshing the context at the end of the destructor ? What was the issue there ?